projects/packages/connection/src/webhooks/class-authorize-redirect.php

```diff
@@ -56,8 +57,12 @@ function ( $domains ) {
 			exit;
 		}
 
+		// The user is either already connected, or finished the connection process.
 		if ( $this->connection->is_connected() && $this->connection->is_user_connected() ) {
-			// The user is either already connected, or finished the connection process.
+			if ( class_exists( '\Automattic\Jetpack\Licensing' ) ) {
```

Comment:
Do you think it would make sense to also handle this via a filter here? In order to avoid any reference to the licensing package?

Reply:
Using a filter there would be nice, but it's complicated by the fact that we don't usually initialize the Licensing package. It gets initialized in the Jetpack plugin and in "My Jetpack", that's it.

Connection package doesn't depend on Licensing in Composer, so we would need to check for the `class_exists()` anyway, and then initialize it for every single plugin that relies on Connection package. That's a lot.
And we would still have to mention the Licensing package in Connection.

So I don't see how we could realistically improve it.
May be you have any ideas?

Reply:
Thanks for the explanation, Sergey! 

I also realized that using a generic filter there (in the Connection package redirect logic) is also not a good idea (even if the Licensing package was always initialized) as any plugin could hook into that and redirect users.